extensions-contrib/google-extensions/src/main/java/org/apache/druid/storage/google/GoogleDataSegmentFinder.java

```diff
@@ -135,8 +135,7 @@ public GoogleDataSegmentFinder(
       throw new SegmentLoadingException(e, "IO exception");
     }
     catch (Exception e) {
-      Throwables.propagateIfInstanceOf(e, SegmentLoadingException.class);
-      Throwables.propagate(e);
+      throw Throwables.propagate(e);
```

Comment:
We need to keep the `Throwables.propagateIfInstanceOf`. Removing it changes the behavior to no longer throw `SegmentLoadingException`, which introduces bugs downstream since some callers specifically try to catch that exception.

Reply:
I'll check it 👍